test/expect-tests/stdune/fpath_tests.ml

```diff
@@ -0,0 +1,41 @@
+open! Stdune
+open Dune_tests_common
+open Fpath
+
+let mkdir_p_result_to_dyn = function
+  | Already_exists -> Dyn.Variant ("Already_exists", [])
+  | Already_exists_not_directory ->
+    Dyn.Variant ("Already_exists_not_directory", [])
+  | Already_exists_parent_not_directory parent ->
+    Dyn.Variant ("Already_exists_parent_not_directory", [ Dyn.string parent ])
+  | Created -> Dyn.Variant ("Created", [])
```

Comment:
You can make this to `Fpath` as this universally useful. The name should be `dyn_of_mkdir_p_result`.